modules/drive/cog.py

```diff
@@ -0,0 +1,83 @@
+import nextcord
+from nextcord.ext import commands
+
+import config
+from modules.drive.drive_service import DriveService
+from utils.embedder import embed_success
+from utils.utils import is_email
+
+from ..error.friendly_error import FriendlyError
+
+
+class DriveCog(commands.Cog):
+    """Manage the ESP Google Drive"""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+        self.service = DriveService(folder_id=config.drive_folder_id)
+
+    @nextcord.slash_command(guild_ids=[config.guild_id])
+    async def drive(self, interaction: nextcord.Interaction[commands.Bot]):
+        """This is a base command for all Google Drive commands and is not invoked"""
+        pass
+
+    @drive.subcommand(name="link")
+    async def drive_link(self, interaction: nextcord.Interaction[commands.Bot]):
+        """Get the link to view the ESP Google Drive folder"""
+        folder_link = f"https://drive.google.com/drive/u/0/folders/{config.drive_folder_id}"
+        await interaction.send(
+            embed=embed_success(
+                title=":link: Google Drive Link",
+                description=(
+                    f"<:google_drive:785981940289372170> [Computer Science Resources JCT ESP All Courses]({folder_link})\n\n"
+                    f"Use the {self.drive_grant.get_mention(interaction.guild)} command to add yourself as a manager of the Google Drive."
+                ),
+                url=folder_link,
+            )
+        )
+
+    @drive.subcommand(name="grant")
+    async def drive_grant(self, interaction: nextcord.Interaction[commands.Bot], email: str):
+        """Add a Google account as a manager of the ESP Google Drive
+
+        Args:
+            email: The email address of the Google account to add
+        """
+        await interaction.response.defer(ephemeral=True)
+        # validate email address
+        if not is_email(email):
+            raise FriendlyError(
+                "Invalid email address",
+                interaction,
+                interaction.user,
+                ephemeral=True,
+            )
+        # add manager to Drive
+        if self.service.add_manager(email):
```

Comment:
prefer this structure:
```
if *not* ...:
    raise ...
continue on success path
```

Reply:
On second thought, instead of reporting success/failure by boolean, raise if failed in the function, and try/except in the cog